dev-utils/gen-notice.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+#
+# Generate a NOTICE file for a distribution of this agent. It includes the
+# "./NOTICE.md" file (which includes the licenses of all vendored code), plus
+# the licenses for all included runtime dependencies.  This will also error out
+# if the license for a dep cannot be determined, or if the license isn't one of
+# the known ones.
+#
+# Usage:
+#   ./dev-utils/gen-notice.sh DIST_DIR
+#
+# where DIST_DIR is the distribution directory (the dir that holds the
+# "package.json" and "node_modules/" dir).
+#
+
+if [ "$TRACE" != "" ]; then
+    export PS4='${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
+    set -o xtrace
+fi
+set -o errexit
+set -o pipefail
+
+# ---- support functions
+
+function fatal {
+    echo "$(basename $0): error: $*"
+    exit 1
+}
+
+# ---- mainline
+
+TOP=$(cd $(dirname $0)/../ >/dev/null; pwd)
+DIST_DIR="$1"
+[[ -n "$DIST_DIR" ]] || fatal "missing DIST_DIR argument"
+[[ -f "$DIST_DIR/package.json" ]] || fatal "invalid DIST_DIR: $DIST_DIR/package.json does not exist"
+
+# Directory holding some "license.*.txt" files for inclusion below.
+export LIC_DIR=$(cd $(dirname $0)/ >/dev/null; pwd)
+
+cat $TOP/NOTICE.md
+
+# Emit a Markdown section listing the license for each non-dev dependency
+# in the DIST_DIR. This errors out if a license cannot be found or isn't known.
+cd $DIST_DIR
+npm ls --omit=dev --all --parseable \
+    | node -e '
+        const fs = require("fs")
+        const path = require("path")
+        const knownLicTypes = {
+            "Apache-2.0": true,
+            "BSD-2-Clause": true,
+            "BSD-3-Clause": true,
+            "CC0-1.0": true,
+            "ISC": true,
+            "MIT": true,
+            "WTFPL OR ISC": true // oddball from is-integer package
+        }
+        const licFileNames = [
+            "LICENSE",
+            "LICENSE.txt",
+            "license.md",
+            "LICENSE-MIT",
+            "LICENSE-MIT.txt"
```

Comment:
Is this going to run on linux build machines?  If so (I think?) we'll need the alternate casing of these file names as well or we'll need something more complex that [the`existsSync`](https://github.com/elastic/apm-agent-nodejs/pull/2764/files#diff-57cd11bbaa52fcd7b495d503a57bd99e109b3a489322b031815d7883cfde6928R104) here.